src/providers/wms/qgswmsdataitems.cpp

```diff
@@ -226,13 +226,44 @@ bool QgsWMSConnectionItem::equal( const QgsDataItem *other )
   {
     return false;
   }
-  const QgsWMSConnectionItem *o = dynamic_cast<const QgsWMSConnectionItem *>( other );
-  if ( !o )
+  const QgsWMSConnectionItem *otherConnectionItem = qobject_cast<const QgsWMSConnectionItem *>( other );
+  if ( !otherConnectionItem )
   {
     return false;
   }
 
-  return ( mPath == o->mPath && mName == o->mName );
+  bool samePathAndName = ( mPath == otherConnectionItem->mPath && mName == otherConnectionItem->mName );
+
+  if ( samePathAndName )
+  {
+    // Check if the children are not the same then they are not equal
```

Comment:
When would this be the case?

Reply:
The current implementation has no guarantee that the wms connection path will be unique, when user edit a wms connection params eg URL only,  the connection path and name does not change but there is a possibility that children items (wms collection items) might change. 

That is why I added this check to confirm it is the same connection item, if not it should be updated.

Reply:
what do you think about @nyalldawson?